### PR TITLE
Full localization support

### LINKED ISSUE #2348: GuiWrongMinecraft: Outdated log file name
`GuiWrongMinecraft` tells the user to look in **ForgeModLoader-client-0.log**, but this was renamed to **fml-client-latest.log** in 1.7.2.

[Original report](http://www.minecraftforge.net/forum/index.php/topic,35958.0.html)

### PR DESCRIPTION
**Pull request by Adaptivity**
Implements #2106
Also, fixes #2348

### CODE REVIEW

File: src/main/resources/assets/forge/lang/en_US.lang [hunk 1]
```diff
@@ -24,7 +24,7 @@ forge.configgui.biomeSkyBlendRange.tooltip=Control the range of sky blending for
 forge.configgui.biomeSkyBlendRange=Biome Sky Blend Range
 forge.configgui.clumpingThreshold.tooltip=Controls the number threshold at which Packet51 is preferred over Packet52.
 forge.configgui.clumpingThreshold=Packet Clumping Threshold
-forge.configgui.disableVersionCheck.tooltip=Set to true to disable Forge's version check mechanics. Forge queries a small json file on our server for version information. For more details see the ForgeVersion class in our github.
+forge.configgui.disableVersionCheck.tooltip=Set to true to disable Forge's version check mechanics. Forge queries a small json file on our server for version information. For more details see the ForgeVersion class in our Github.
```
Comment (kashike) [inline src/main/resources/assets/forge/lang/en_US.lang]: If you're changing the capitalisation of "github" here, capitalise properly: GitHub

File: src/main/resources/assets/forge/lang/en_US.lang [hunk 2]
```diff
@@ -140,8 +140,56 @@ fml.configgui.tooltip.undoChanges=Undo Changes
...
+fml.menu.modinfo.version=Version:
+fml.menu.modinfo.modState=Mod State:
+fml.menu.modinfo.noInformation=No mod information found
+fml.menu.modinfo.nomcmodFile=Ask your mod author to provide a mod mcmod.info file
+fml.menu.modinfo.changes=Changes:
+fml.menu.modlist=Mod List
+fml.menu.modoptions=Mod Options...
 fml.menu.mods=Mods
+fml.menu.modsLoadedActiveSingular=%d mod%s loaded, %d mod%s active
+fml.menu.modsLoadedPlural=s
```
Comment (kashike) [inline src/main/resources/assets/forge/lang/en_US.lang]: Not all languages operate like this, entire string should be translated for singular/plural.

Comment (chylex) [inline src/main/resources/assets/forge/lang/en_US.lang]: Some languages require multiple states of plural (2-4, 3-10, 13-19, 20+, ending with 1... just a couple examples I know of or found), I don't think there is a good way of handling it without major changes.

Comment (Adaptivity) [inline src/main/resources/assets/forge/lang/en_US.lang]: I think he mean that I should add one full entry for singular and another one for plural. Now I need to find out how to do this... (my programming skills are very simple)

Comment (chylex) [inline src/main/resources/assets/forge/lang/en_US.lang]: Yea, just saying that there are different types of plural in languages having singular and plural in this way doesn't fix the problem, just postpones it.

File: src/main/java/net/minecraftforge/fml/common/FMLCommonHandler.java [hunk 3]
```diff
@@ -228,7 +229,7 @@ public void computeBranding()
             }
             int tModCount = Loader.instance().getModList().size();
             int aModCount = Loader.instance().getActiveModList().size();
-            brd.add(String.format("%d mod%s loaded, %d mod%s active", tModCount, tModCount!=1 ? "s" :"", aModCount, aModCount!=1 ? "s" :"" ));
+            brd.add(StatCollector.translateToLocalFormatted("fml.menu.modsLoadedActiveSingular", tModCount, tModCount!=1 ? StatCollector.translateToLocalFormatted("fml.menu.modsLoadedPlural") :"", aModCount, aModCount!=1 ? StatCollector.translateToLocalFormatted("fml.menu.modsActivePlural") :"" ));
             brandings = brd.build();
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/fml/common/FMLCommonHandler.java]: I would avoid anything in the main menu like this, it gives people a vector for vanity/annoyance to the end user by allowing them to clutter the main menu. We already give them 1 line worth of branding don't want to open up more.

Comment (Adaptivity) [inline src/main/java/net/minecraftforge/fml/common/FMLCommonHandler.java]: So, should I just simply hardcode this back in?

**Comment by cpw** [comment]
Comment: as I asked previously don't localize the log files, please! It's bad enough today, but it'll be impossible to support anyone with problems if we can't even read my log messages anymore.

The only elements that should be localized are the on screen ones. If the messages are shared, do NOT localize them, thanks!

**Comment by Adaptivity** [comment]
Hmm... can you link me which lines are used in log files? I guess *font.drawString* and *this.drawCenteredString* strings are shown only on screen or I'm wrong?

File: src/main/resources/assets/forge/lang/en_US.lang (hunk 2 above)
Comment (immibis) [inline src/main/resources/assets/forge/lang/en_US.lang]: There's no way to fix that unless you want to put arbitrary programs in language files.
(Which I think some other non-Minecraft project actually did)

**Comment by Actuarius** [comment]
@williewillus added labels [Cleanup]

**Pull request closed without merging**